core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -324,6 +331,12 @@ private static void writeSetSnapshotRef(MetadataUpdate.SetSnapshotRef update, Js
     JsonUtil.writeLongFieldIf(update.maxRefAgeMs() != null, MAX_REF_AGE_MS, update.maxRefAgeMs(), gen);
   }
 
+  private static void writeRemoveSnapshotRef(
+      MetadataUpdate.RemoveSnapshotRef update,
+      JsonGenerator gen) throws IOException {
```

Comment:
Nit: if this could be on one line, possibly with the `throws IOException` on its own line if it doesn't all fit on one line, that would be great / fit what is done elsewhere.

But I wouldn't block on this PR for that.

```java
  private static void writeRemoveSnapshotRef(MetadataUpdate.RemoveSnapshotRef update, JsonGenerater gen)
      throws IOException {
    gen.writeStringField(REF_NAME, update.name());
  }
```

Reply:
`writeSetDefaultSortOrder` is written in this style iirc.